src/IconButton/index.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cc from "classcat";
+import iconSelection from "src/icons/selection.json";
+
+export const VALID_ICON_NAMES = iconSelection.icons.map(
+  (icon) => icon.properties.name
+);
+
+/**
+ * Narmi style Icon Button.
+ *
+ * Icon Button renders a button with a central Narmi icon
+ *
+ * This component supports rest props; any additional props on icon button will be
+ * passed through to button element.
+ */
+const IconButton = ({
+  disabled = false,
+  kind = "plain",
+  size = "m",
+  testId = "nds-icon-button",
+  label,
+  name,
+  onClick = () => {},
+  className,
+  ...otherProps
+}) => {
+  return (
+    <button
+      tabIndex={0}
+      onClick={onClick}
+      {...otherProps}
```

Comment:
We generally avoid spread props on NDS components except for form elements. 

The idea is that we want to prevent `style` and `className` from being passed - if there's a styling change we need to make in product, we should work with design to come up with a standard variant instead of accepting arbitrary styles.

In this case, your `className` prop comes after spread props, so we may be fine if `style` is plucked from `...otherProps`